packages/gatsby/src/datastore/types.ts

```diff
@@ -9,6 +9,11 @@ export interface ILmdbDatabases {
   nodesByType: Database<NodeId, NodeType>
 }
 
+export interface IQueryResult {
+  entries: Iterable<IGatsbyNode>
+  totalCount: number | (() => Promise<number>)
+}
```

Comment:
`totalCount` will be expensive for `lmdb` case, so we support a `thunk` option too.